File: pypp_cli/test_dir/python/src/file_io/test_is_here.py
from pypp_python import res_dir
import os


def test_is_here_fn():
    print("FILE IO TEST IS HERE RESULTS:")
    # NOTE: This works for the C++ only if the executable is in the first level of
    # the build/ dir
    resources_dir: str = res_dir()
    file: str = os.path.join(resources_dir, "test_is_here.txt")
    if os.path.isfile(file):
        print("test_is_here.txt file exists")
    else:
        raise Exception("test_is_here.txt file does not exist")

File: pypp_cli/test_dir/cpp/src/file_io/test_is_here.cpp
#include "src/file_io/test_is_here.h"
#include "exceptions/exception.h"
#include "py_str.h"
#include "pypp_os.h"
#include "pypp_resources.h"
#include "pypp_util/print.h"

namespace me {
void test_is_here_fn() {
    pypp::print(pypp::PyStr("FILE IO TEST IS HERE RESULTS:"));
    pypp::PyStr resources_dir = pypp::res_dir();
    pypp::PyStr file =
        pypp::os::path::join(resources_dir, pypp::PyStr("test_is_here.txt"));
    if (pypp::os::path::isfile(file)) {
        pypp::print(pypp::PyStr("test_is_here.txt file exists"));
    } else {
        throw pypp::Exception(
            pypp::PyStr("test_is_here.txt file does not exist"));
    }
}

} // namespace me